test/e2e/app-dir/index.test.ts

```diff
@@ -1494,57 +1501,89 @@ describe('app dir', () => {
       )
     })
 
-    // TODO-APP: This is disabled for development as the error overlay needs to be reworked.
-    ;(isDev ? describe.skip : describe)('error component', () => {
+    describe('error component', () => {
       it('should trigger error component when an error happens during rendering', async () => {
-        const browser = await webdriver(next.url, '/error/clientcomponent')
-        await browser
-          .elementByCss('#error-trigger-button')
-          .click()
-          .waitForElementByCss('#error-boundary-message')
-
-        expect(
-          await browser.elementByCss('#error-boundary-message').text()
-        ).toBe('An error occurred: this is a test')
-      })
-
-      it('should allow resetting error boundary', async () => {
-        const browser = await webdriver(next.url, '/error/clientcomponent')
-
-        // Try triggering and resetting a few times in a row
-        for (let i = 0; i < 5; i++) {
+        const browser = await webdriver(next.url, '/error/client-component')
+        await browser.elementByCss('#error-trigger-button').click()
+
+        if (isDev) {
+          expect(await hasRedbox(browser)).toBe(true)
+          console.log('getRedboxHeader', await getRedboxHeader(browser))
+          // expect(await getRedboxHeader(browser)).toMatch(/An error occurred: this is a test/)
+        } else {
           await browser
-            .elementByCss('#error-trigger-button')
-            .click()
-            .waitForElementByCss('#error-boundary-message')
-
           expect(
-            await browser.elementByCss('#error-boundary-message').text()
+            await browser
+              .waitForElementByCss('#error-boundary-message')
+              .elementByCss('#error-boundary-message')
+              .text()
           ).toBe('An error occurred: this is a test')
-
-          await browser
-            .elementByCss('#reset')
-            .click()
-            .waitForElementByCss('#error-trigger-button')
-
-          expect(
-            await browser.elementByCss('#error-trigger-button').text()
-          ).toBe('Trigger Error!')
         }
       })
 
-      it('should hydrate empty shell to handle server-side rendering errors', async () => {
+      it('should fallback default error boundary when no error component specified', async () => {
```

Comment:
The fallback error boundary should never show in development as the error overlay boundary should have caught it 🤔

Reply:
Yes the tests fallback to the global one only in prod, use the dev overlay in dev mode. need to rewrite the test case name